pkg/storage/kubernetes/ipam.go

```diff
@@ -313,6 +313,7 @@ func newLeaderElector(clientset *kubernetes.Clientset, namespace string, podName
 		LeaseDuration: time.Duration(leaseDuration) * time.Millisecond,
 		RenewDeadline: time.Duration(renewDeadline) * time.Millisecond,
 		RetryPeriod:   time.Duration(retryPeriod) * time.Millisecond,
+		ReleaseOnCancel: true,
```

Comment:
I think the commit msg should explicitly quote the name of the attribute - `ReleaseOnCancel`.

Furthermore, while I think this is an amazing find, and in the harmless, I see in the [LE docs](https://github.com/kubernetes/client-go/blob/v0.22.2/tools/leaderelection/leaderelection.go#L150)  that `If you set this to true, you must ensure all code guarded by this lease has successfully completed prior to cancelling the context` . 

I get a bit confused by their wording, especially with "successfully completed." Can we guarantee that ?..

Reply:
I agree with @maiqueb, the leader lease should be held till the update to ippools CRD is completed. If cancellation happens due to timeout before update is complete, duplicate IP allocation might occur.

Reply:
> I get a bit confused by their wording, especially with "successfully completed." Can we guarantee that ?..
I believe we can. The LE context being "canceled" for ```le.Run()``` should only happen when case <-leader is guaranteed to have completed or we can be certain its not the case being triggered. Current master, the LE context is only called when case <- leader is finished so we are certain it has completed, therefore its good to go into master now.
Any other PRs (like #142 ) need, we need to keep this in mind, and I think I have.